### PR TITLE
Editor code: tidy up BOOLs,NULLs and overrides pt2.

### PR DESCRIPTION
**Pull request by nemerle**
A few 'typedefs' replaced by 'using's
This shouldn't have any functional changes at all, just c++17 modernization
It's a part 2 of a split #2847

*Note to reviewers this is a split PR meant to ease reviewing, but sadly I cannot say if this or any other part will compile correctly in isolation.*

Signed-off-by: Nemerle <nemerle5+git@gmail.com>

### CODE REVIEW

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by hultonha — APPROVED** [review]
LGTM! 🙂

File: Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp [hunk 1]
```diff
@@ -165,8 +165,8 @@ namespace ProjectSettingsTool
             if (editContext)
             {
                 editContext->Class<AndroidSplashscreens>("Splashscreens", "All splashscreen overrides for Android.")
-                    ->DataElement(0, &AndroidSplashscreens::m_landscapeSplashscreens)
-                    ->DataElement(0, &AndroidSplashscreens::m_portraitSplashscreens)
+                    ->DataElement(nullptr, &AndroidSplashscreens::m_landscapeSplashscreens)
```
Comment (hultonha) [inline Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp:168]: Actually sorry I just noticed I think this is wrong, this function is an overload and I think this should really be `AZ::Edit::UIHandlers::Default` not `nullptr` (see `EditContext.h` for the overloads)

**Review by hultonha — CHANGES_REQUESTED** [review]
Sorry was nearly perfect, just one thing to change (this might crop up in a few more places too potentially)

File: Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp (hunk 1 above)
Comment (nemerle) [reply Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp:168]: Yup, You're 100% that this is incorrect, will fix in ~1h

Comment (nemerle) [reply Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp]: Fixed in f2c482b

Comment (hultonha) [reply Code/Editor/Plugins/ProjectSettingsTool/PlatformSettings_Android.cpp]: Awesome thanks! 🙂

**Review by hultonha — APPROVED** [review]

**Pull request merged**